types/node/index.d.ts

```diff
@@ -2181,19 +2185,18 @@ declare module "url" {
         pathname?: string;
         port?: string | number;
         protocol?: string;
-        query?: string | null | { [key: string]: string | string[] };
+        query?: string | null | ParsedUrlQuery;
         search?: string;
         slashes?: boolean;
     }
 
-    export interface Url extends UrlObject {
+    export interface OutputUrlObject extends InputUrlObject {
         port?: string;
-        query?: any;
     }
 
-    export function parse(urlStr: string, parseQueryString?: boolean, slashesDenoteHost?: boolean): Url;
+    export function parse(urlStr: string, parseQueryString?: boolean, slashesDenoteHost?: boolean): OutputUrlObject;
     export function format(URL: URL, options?: URLFormatOptions): string;
-    export function format(urlObject: UrlObject | string): string;
+    export function format(urlObject: InputUrlObject | string): string;
```

Comment:
For clarification on "input" and "output" types: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/18766#discussion_r145259190